### ISSUE TITLE
Setting layer-wise learning rate doens't work

### ISSUE THREAD

**Issue by zhiyuanpeng** [PyTorchLightning/pytorch-lightning#11269]
## 🐛 Bug

Hi, I use the following code to set `learning rate per layer` but it seems that the `lr` in `optimizer_grouped_parameters` doesn't work. I set `lr=1` in `optimizer_grouped_parameters` and the result should be very worse but the result doesn't change. The result only varies with different global `self.hparams.lr`s.
```
def configure_optimizers(self):
        """Choose what optimizers and learning-rate schedulers to use in your optimization.
        Normally you'd need one. But in the case of GANs or similar you might have multiple.

        See examples here:
            https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers
        """
        no_decay = ["bias", "LayerNorm.weight"]
        optimizer_grouped_parameters = [
            {
                "params": [
                    p
                    for n, p in self.named_parameters()
                    if not any(nd in n for nd in no_decay)
                ],
                "lr": 1.0,
                "weight_decay": self.hparams.weight_decay,
            },
            {
                "params": [
                    p
                    for n, p in self.named_parameters()
                    if any(nd in n for nd in no_decay)
                ],
                "weight_decay": 0.0,
            }
        ]
        optimizer = torch.optim.AdamW(
            params=optimizer_grouped_parameters, lr=self.hparams.lr, eps=self.hparams.adam_eps
        )
        return optimizer
```
### To Reproduce

Use the above piece of code `configure_optimizers`. Setting `lr` in `optimizer_grouped_parameters` doesn't change the results.

### Expected behavior

Setting `lr` in `optimizer_grouped_parameters` should change the results.

### Environment

* CUDA:
        - GPU:
                - Tesla V100-SXM2-16GB
                - Tesla V100-SXM2-16GB
                - Tesla V100-SXM2-16GB
                - Tesla V100-SXM2-16GB
        - available:         True
        - version:           10.2
* Packages:
        - numpy:             1.19.5
        - pyTorch_debug:     False
        - pyTorch_version:   1.10.0+cu102
        - pytorch-lightning: 1.5.5
        - tqdm:              4.62.3
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.8.10
        - version:           #135-Ubuntu SMP Fri Jan 19 11:48:36 UTC 2018

### Additional context

No
